source/mesh_deformation/interface.cc

```diff
@@ -464,6 +501,116 @@ namespace aspect
 
 
 
+    template <int dim>
+    AffineConstraints<double> MeshDeformationHandler<dim>::make_initial_constraints()
+    {
+      AssertThrow(this->get_parameters().mesh_deformation_enabled, ExcInternalError());
+
+      AffineConstraints<double> initial_deformation_constraints;
+
+      // Now construct the mesh displacement constraints
+      initial_deformation_constraints.reinit(mesh_locally_relevant);
+
+      // mesh_velocity_constraints can use the same hanging node
+      // information that was used for mesh_vertex constraints.
+      initial_deformation_constraints.merge(mesh_vertex_constraints);
```

Comment:
could you just copy? the lhs object is empty here

Reply:
If I just replace `.merge` by `.copy_from` I get the following error:
```
00: --------------------------------------------------------
100: An error occurred in line <1949> of file </home/rene/software/deal.II-dev/include/deal.II/lac/affine_constraints.h> in function
100:     void dealii::AffineConstraints<number>::add_line(dealii::AffineConstraints<number>::size_type) [with number = double; dealii::AffineConstraints<number>::size_type = unsigned int]
100: The violated condition was: 
100:     sorted == false
100: Additional information: 
100:     (none)
100: 
100: Stacktrace:
100: -----------
100: #0  /home/rene/software/aspect-debug/aspect: dealii::AffineConstraints<double>::add_line(unsigned int)
100: #1  /home/rene/software/deal.II-dev/lib/libdeal_II.g.so.9.3.0-pre: void dealii::VectorTools::interpolate_boundary_values<2, 2, dealii::DoFHandler, double>(dealii::Mapping<2, 2> const&, dealii::DoFHandler<2, 2> const&, std::map<unsigned int, dealii::Function<2, double> const*, std::less<unsigned int>, std::allocator<std::pair<unsigned int const, dealii::Function<2, double> const*> > > const&, dealii::AffineConstraints<double>&, dealii::ComponentMask const&)
100: #2  /home/rene/software/deal.II-dev/lib/libdeal_II.g.so.9.3.0-pre: void dealii::VectorTools::interpolate_boundary_values<2, 2, dealii::DoFHandler, double>(dealii::Mapping<2, 2> const&, dealii::DoFHandler<2, 2> const&, unsigned int, dealii::Function<2, double> const&, dealii::AffineConstraints<double>&, dealii::ComponentMask const&)
100: #3  /home/rene/software/aspect-debug/aspect: aspect::MeshDeformation::MeshDeformationHandler<2>::make_initial_constraints()
```

And I get the same error if I use the copy-constructor.

Reply:
a call to compress() should fix this error (but I think you can leave things as is).